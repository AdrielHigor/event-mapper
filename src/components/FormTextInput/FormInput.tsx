import React from "react";
import "./FormInput.css";

interface IFormTextInput {
  label: string;
  id: string;
  onChange?: React.ChangeEventHandler<HTMLInputElement> | undefined;
  inputType?: React.HTMLInputTypeAttribute | undefined;
  isValid?: boolean;
}

const FormInput = ({
  inputType,
  label,
  id,
  isValid,
  onChange,
}: IFormTextInput) => (
  <div className="input-container">
    <label
      htmlFor={id}
      className={`form-label ${!isValid && "form-label-error"}`}
    >
      {label}
    </label>
    <input
      onChange={onChange}
      id={id}
      className={`form-input ${!isValid && "form-input-error"}`}
      type={inputType}
    ></input>
    {!isValid && <p className="form-error-text">Este campo é obrigatório!</p>}
  </div>
);

export default FormInput;
