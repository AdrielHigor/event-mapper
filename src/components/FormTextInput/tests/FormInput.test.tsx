import renderer from "react-test-renderer";
import FormInput, { IFormTextInput } from "../FormInput";

describe("FormInput", () => {
  const defaultProps: IFormTextInput = {
    label: "test",
    id: "test",
    inputType: "text",
  };

  const wrapper = (options = defaultProps) => {
    return renderer.create(<FormInput {...options} />);
  };

  it("renders correctly", () => {
    const component = wrapper({ ...defaultProps, isValid: true });
    expect(component).toMatchSnapshot();
  });

  it("renders correctly with error styling when isValid prop is false", () => {
    const component = wrapper({ ...defaultProps, isValid: false });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls onChange handler when input value is changed", () => {
    const mockOnChange = jest.fn();
    const component = wrapper({ ...defaultProps, onChange: mockOnChange });

    const input = component.root.findByType("input");
    input.props.onChange({ target: { value: "John Doe" } });
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
