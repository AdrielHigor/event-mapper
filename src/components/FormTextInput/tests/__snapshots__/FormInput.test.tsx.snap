// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders correctly 1`] = `
<div
  className="input-container"
>
  <label
    className="form-label false"
    htmlFor="test"
  >
    test
  </label>
  <input
    className="form-input false"
    id="test"
    type="text"
  />
</div>
`;

exports[`FormInput renders correctly with error styling when isValid prop is false 1`] = `
<div
  className="input-container"
>
  <label
    className="form-label form-label-error"
    htmlFor="test"
  >
    test
  </label>
  <input
    className="form-input form-input-error"
    id="test"
    type="text"
  />
  <p
    className="form-error-text"
  >
    Este campo é obrigatório!
  </p>
</div>
`;
